test_responses.py

```diff
@@ -937,3 +937,21 @@ def test_cookies_from_headers():
     for k, v in cookiejar.items():
         assert isinstance(v, str)
         assert v == expected[k]
+
+
+def test_cookie_attribute_handling():
+    expected_domains = [".stackoverflow.com"]
+    expected_paths = ["/"]
+    cookie_attributes = [
+        "prov=c83dc7dc-e0b9-xxxx-xxxx-xxxxxxxxxxxx",
+        "domain=.stackoverflow.com",
+        "expires=Fri, 01-Jan-2055 00:00:00 GMT",
+        "path=/",
+        "HttpOnly",
+    ]
+    cookie_text = "; ".join(cookie_attributes)
+    headers = {"set-cookie": cookie_text}
+    cookiejar = responses._cookies_from_headers(headers)
```

Comment:
Is there anyway we can test cookie handling through a public interface instead of via protected methods?